roadmap/implementors-guide/src/types/candidate.md

```diff
@@ -0,0 +1,150 @@
+# Candidate Types
+
+Para candidates are some of the most common types, both within the runtime and on the Node-side.
+
+In a way, this entire guide is about these candidates: how they are scheduled, constructed, backed, included, and challenged.
+
+This section will describe the base candidate type, its components, and abridged counterpart.
+
+## CandidateReceipt
+
+This is the base receipt type. The `GlobalValidationSchedule` and the `LocalValidationData` are technically redundant with the `relay_parent`, which uniquely describes the a block in the blockchain from whose state these values are derived. The [`AbridgedCandidateReceipt`](#abridgedcandidatereceipt) variant is often used instead for this reason.
+
+However, the full CandidateReceipt type is useful as a means of avoiding the implicit dependency on availability of old blockchain state. In situations such as availability and approval, having the full description of the candidate within a self-contained struct is convenient.
+
+```rust
+/// All data pertaining to the execution of a para candidate.
+struct CandidateReceipt {
+	/// The ID of the para this is a candidate for.
+	para_id: ParaId,
+	/// The hash of the relay-chain block this should be executed in
+	/// the context of.
+	relay_parent: Hash,
+	/// The head-data produced as a result of execution.
+	head_data: HeadData,
+	/// The collator's sr25519 public key.
+	collator: CollatorId,
+	/// Signature on blake2-256 of components of this receipt:
+	/// The parachain index, the relay parent, the head data, and the pov_hash.
+	signature: CollatorSignature,
+	/// The blake2-256 hash of the PoV.
+	pov_hash: Hash,
+	/// The global validation schedule.
+	global_validation: GlobalValidationSchedule,
```

Comment:
Would it be worth embedding an `AbridgedCandidateReceipt` instead of embedding the fields directly?

Reply:
I didn't do this in the past for some reason, but can't remember why. Seems fine to me.

I'd also be fine with renaming `CandidateReceipt` to `FullCandidateReceipt` or `FreeStanding*` or `Executable*` or something, since the abridged version is by far more common.

Reply:
The only argument I can think of against it is that embedding the fields has a flatter structure which makes instantiation for tests easier. However, in the past we didn't have a `Default` implementation for these types but do now, which makes this less of a concern.

Reply:
I'd 👍 a simultaneous rename of CandidateReceipt -> FullCandidateReceipt, AbridgedCandidateReceipt -> CandidateReceipt, and embedding the latter into the former.

Reply:
Oh, just saw that you already embedded it. Cool!

Reply:
Let's do the rename in a follow-up